### ISSUE TITLE
Empty tags post deposition

### ISSUE THREAD

**Issue by rsiegl** [uwbmrb/BMRBDep#134]
I am not sure if these steps are done after deposition or not so this may be irrelevant.

I have noticed that after depositing an entry most every loop has the following tags with '.' : ID, Entry_ID, (loop name)_ID for example Citation_ID. When you deposit an entry from an existing BMRB entry by providing the number some of these tags (the ones that were pre-filled) are filled out, but with the Entry_ID of the old entry.

I have also noticed that the _Upload_data.Data_file_ID has odd behavior. For example I only have one data file listed yet this tag is set to 2. 

The entity section is also missing the _Entity_comp_index.ID and _Entity_poly_seq loops.

**Comment by dmaziuk** [comment]
Some are done post-deposition in ADIT: sequence loops and filling in IDs. I believe old entry IDs as well as some other tags are stripped off on upload in ADIT. (Uploaded file names in several places and one or two others that I can't remember off the top of my head).

**Comment by jonwedell** [comment]
> I have noticed that after depositing an entry most every loop has the following tags with '.' : ID, Entry_ID, (loop name)_ID for example Citation_ID. When you deposit an entry from an existing BMRB entry by providing the number some of these tags (the ones that were pre-filled) are filled out, but with the Entry_ID of the old entry.

@dmaziuk  - Do I need to set the ID tags/pointers, or will the existing pipeline code do that? I remember discussing it but don't remember the outcome.

I can easily set the Entry_ID tags on deposition.

Old file names will get set, but will be invalid and will be have to re-selected by the depositor - unless they re-upload files with those same names.

> 
> I have also noticed that the _Upload_data.Data_file_ID has odd behavior. For example I only have one data file listed yet this tag is set to 2.

Fixed.

**Comment by rsiegl** [comment]
Gotcha. So ideally it would look as it does below then?

![1](https://user-images.githubusercontent.com/44450691/56388773-80597300-61ed-11e9-8a4a-517eadd71d7f.png)


I can confirm this is fixed also.

**Comment by dmaziuk** [comment]
Right now there is a separate piece of code that runs on the output of ADIT-NMR and, among other things, sets all `Entry_ID` tags to `NEED_ACC_NUM`

If you support generation of "preview" NMR-STAR file, I recommend using that instead of e.g. uuid -- this is the one annotators expect to see.

The saveframe pointers are much much uglier in ADIT (it has no saveframes) so the code that resolves those won't be usable as-is.  The final piece generates local IDs for each saveframe and populates `_whatever_ID` tags based on the corresp. `_whatever_label` tags. That, and the function for renumbering local IDs, also exists in the interactive validator. 

I'd say either leave it to the annotators or do it so you generate a "nice" output file (e.g. the preview file that isn't annotated). Existing code shells out to a java program that does some parts of what you're asking... I'd rather get rid of that.

**Comment by jonwedell** [comment]
> If you support generation of "preview" NMR-STAR file, I recommend using that instead of e.g. uuid -- you should put in something, you don't have a BMRB ID yet. `NEED_ACC_NUM` is what the annotators expect to see.

I actually don't have an (exposed) way in the interface for users to generate a preview file yet, though there is a hidden one for development. Currently it spits out a file with the uuid in the saveframe Entry_ID tags (and '.' in the loop Entry_ID tags), but updating the code to replace that with NEED_ACC_NUM would actually be a lot of work (the TypeScript NMR-STAR library I wrote is much more spartan than the Python one). Even if I expose this to users in the future, these "preview" files would only be used to start new depositions, in which case all the Entry_ID tags are stripped anyway. So I'm thinking I can get away with leaving this as is, unless you feel very strongly about it.

What I have just implemented is that during deposition all the Entry_ID tags get set to the actual BMRB ID as soon as one is assigned, so annotators shouldn't ever need to see the `NEED_ACC_NUM` in the future with this system.



> That, and the function for renumbering local IDs, also exists in the interactive validator.

Okay, I think my plan will be for the annotators to assign the `ID` tags using the validator function for now, but I will update the PyNMR-STAR library to have a function for doing this eventually, and then I will do it during the deposition step. It will be nice to have, but isn't mandatory for the first release. I'll make an issue to track this.

**Comment by jonwedell** [comment]
> but with the Entry_ID of the old entry.

This is also fixed now (1f2bbb6). The jumpstart entry Entry_ID's were leaking into the merged entry, but just in the loops.

**Comment by rsiegl** [comment]
I cannot verify this one until that issue with the experiment file section is fixed Issue #137 (There's nothing in the drop down so I cannot deposit since that is a required tag). Once that is resolved I can check this along with Issue #82.

**Comment by jonwedell** [comment]
If you set "Time-domain data provided" to "No" for all loops it should disappear. (You may need to reload the page too.) But no hurry either, I can update when that is finished.

**Comment by jonwedell** [comment]
I've pushed a new version of the code, it should work with existing depositions in progress, though you will need to reload the cached entry using the special "Reload entry" button from the side menu, and then reload the page. (Or end session, and then resume session from the link in the email.)

**Comment by rsiegl** [comment]
I tried to open up some existing depositions and got the following error and they do not load.

![2](https://user-images.githubusercontent.com/44450691/56616023-1a505f80-65e2-11e9-846b-2f32090c0083.png)

I also tried to open a new entry. I was not able to do that either and got this error.

![1](https://user-images.githubusercontent.com/44450691/56616060-2d632f80-65e2-11e9-834c-df01d0bee42b.png)

**Comment by jonwedell** [comment]
> I tried to open up some existing depositions and got the following error and they do not load.

Argh, sorry. Fixed.

**Comment by rsiegl** [comment]
I can access entries now, but I cannot deposit them. The "A network or server exception ocurred" error pops up when I try to.

**Comment by jonwedell** [comment]
Can you e-mail me the deposition ID?

**Comment by jonwedell** [comment]
Ah, I know what the problem is.

**Comment by jonwedell** [comment]
Should be fixed now.

**Comment by rsiegl** [comment]
Huh I still get that error. I emailed you the entry ID's of a few entries I have open.

**Comment by rsiegl** [comment]
I downloaded an entry earlier to see if that issue about the old entry ID's leaking into the new entry was fixed and it is. Now the .ID, .Entry_ID and .(whatever section)_list_ID in loops are no longer there. I am not sure if that is the behavior you were looking for, but I included an example below along with the old version.

Current version:
![1](https://user-images.githubusercontent.com/44450691/56833368-f08e7700-6833-11e9-8283-e40ee691e251.png)

Old entry version:
![2](https://user-images.githubusercontent.com/44450691/56833383-f8e6b200-6833-11e9-87b7-e91a70a03dc0.png)
